UITests/UITests.App/App.AppService.xaml.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Toolkit.Mvvm.Messaging;
+using UITests.App.Pages;
+using Windows.ApplicationModel.Activation;
+using Windows.ApplicationModel.AppService;
+using Windows.ApplicationModel.Background;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace UITests.App
+{
+    /// <summary>
+    /// This file contains part of the app related to the AppService for communication between this test host and the test harness processes.
+    /// </summary>
+    public sealed partial class App
+    {
+        private AppServiceConnection _appServiceConnection;
+        private BackgroundTaskDeferral _appServiceDeferral;
+
+        protected override void OnBackgroundActivated(BackgroundActivatedEventArgs args)
+        {
+            base.OnBackgroundActivated(args);
+            IBackgroundTaskInstance taskInstance = args.TaskInstance;
+            AppServiceTriggerDetails appService = taskInstance.TriggerDetails as AppServiceTriggerDetails;
+            _appServiceDeferral = taskInstance.GetDeferral();
+            taskInstance.Canceled += OnAppServicesCanceled;
+            _appServiceConnection = appService.AppServiceConnection;
+            _appServiceConnection.RequestReceived += OnAppServiceRequestReceived;
+            _appServiceConnection.ServiceClosed += AppServiceConnection_ServiceClosed;
+        }
+
+        private async void OnAppServiceRequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            AppServiceDeferral messageDeferral = args.GetDeferral();
+            ValueSet message = args.Request.Message;
+            string cmd = message["Command"] as string;
+
+            try
+            {
+                // Return the data to the caller.
+                if (cmd == "Start")
+                {
+                    var pageName = message["Page"] as string;
+
+                    Log.Comment("Received request for Page: {0}", pageName);
+
+                    ValueSet returnMessage = new ValueSet();
+
+                    // We await the OpenPage method to ensure the navigation has finished.
+                    if (await WeakReferenceMessenger.Default.Send<RequestPageMessage>(new (pageName)))
```

Comment:
Just realized we _can_ actually simplify this more, since the type argument can be inferred 😄
Or at least, we have a less verbose variant, without angle brackets:

```suggestion
                    if (await WeakReferenceMessenger.Default.Send(new RequestPageMessage(pageName)))
```

Slightly shorter? Also this version doesn't even require C# 9, if that matters.

Reply:
That's a good point. Maybe I do that vs. forcing the C# 9 on the project earlier...